gufe/mapping/atom_mapping.py

```diff
@@ -0,0 +1,53 @@
+# This code is part of gufe and is licensed under the MIT license.
+# For details, see https://github.com/OpenFreeEnergy/gufe
+
+import abc
+from collections.abc import Mapping, Iterable
+
+
+import gufe
+from .componentmapping import ComponentMapping
+
+
+class AtomMapping(ComponentMapping, abc.ABC):
+    _molA: gufe.Component
+    _molB: gufe.Component
+
+    """A mapping between two different atom-based Components"""
+
+    @property
+    def molA(self) -> gufe.Component:
+        """A copy of the first Component in the mapping"""
+        return self._molA
+
+    @property
+    def molB(self) -> gufe.Component:
+        """A copy of the second Component in the mapping"""
+        return self._molB
+
+    @property
+    @abc.abstractmethod
+    def molA_to_molB(self) -> Mapping[int, int]:
+        """Maps the index of an item from Component A onto Component B
```

Comment:
Or we switch completely to calling them "components" (I think that'd be worse for users, but maybe better for devs)

Reply:
I think somewhere I realised Molecule is a loaded term (refers to class names elsewhere, and/or is a single connected molecule) Component less so.

Reply:
Component isn't going to float well with users (it doesn't really mean much). Subclasses of this will be interfaced with by users, so we should probably should come up with definite semantics here.

Reply:
It felt to me like the `Component` idea went pretty well when I explained it in my talk in Amsterdam, so I think users will get it.

I agree that `molA` is misleading when the thing should be `componentA`.

Reply:
I disagree that `Component` is reasonable for users, especially given that you always do perturbations on molecules (you won't even just perturb the entire solvent for example, that mapping is purely on the ligand). That being said I also just want this merged. I'd prefer we just stick to _something_ for now, and then I can just poll all the FE users in our lab to tell us why neither name makes sense.

Reply:
As long as (up to) a biological assembly can be referred to as a mol_A that's fine.

Reply:
Right because we could just throw in an entire protein component and just expect this to fall within this map? (waters and all)

I'm not sure that's super digestible from a user perspective, but I guess we've already bitten the bullet (or at least there's no easy way around this).

Reply:
Ok I've gone with Component, if everyone hates it we change it in v2

Reply:
We should do "user testing" as part of our v2 goal.